test/jdk/java/awt/event/KeyEvent/FunctionKeyTest.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 1998, 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.BorderLayout;
+import java.awt.Button;
+import java.awt.Color;
+import java.awt.Event;
+import java.awt.Frame;
+import java.awt.Label;
+import java.awt.TextArea;
+import javax.swing.SwingUtilities;
+
+/*
+ * @test
+ * @bug 4011219
+ * @summary Test for function key press/release received by Java client.
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @run main/manual FunctionKeyTest
+ */
+
+public class FunctionKeyTest {
+    private static final String INSTRUCTIONS = """
+            Press and release function keys F11 and F12.
+            Look at the test window:
+               On KeyPress:   'e.id=403' is printed
+               On KeyRelease: 'e.id=404' is printed
+            If the above is true, then click Pass, else click Fail.
+            """;
+
```

Comment:
This can be converted to automated test using robot's Key events `(VK_F11, VK_F12)`.

Reply:
Sure, converted.

Reply:
Automated test changes look good. 
`handleEvent` and `keyDown` is marked for deprecation, should it be replaced with `processEvent()`?

Reply:
I'm not sure that in this case we will test the same thing that the test was written for.
So I've left things as they are intentionally.